backend/src/main/java/com/huellapositiva/domain/service/EmailCommunicationService.java

```diff
@@ -69,4 +69,26 @@ public void sendSubmittedProposalRevision(ProposalRevisionEmail proposalRevision
         Email email = Email.createFrom(proposalRevisionEmail, emailTemplate, from);
         emailService.sendEmail(email);
     }
+
+
+
+    public void sendRecoveryPasswordEmail(EmailRecoveryPassword emailRecoveryPassword) {
+        try {
+            EmailTemplate emailTemplate = templateService.getRecoveryEmailTemplate(emailRecoveryPassword.getHash());
+            Email email = Email.createFrom(emailRecoveryPassword, emailTemplate, from);
+            emailService.sendEmail(email);
+        } catch (RuntimeException ex) {
```

Comment:
Do not throw or catch RuntimeExceptions unless necessary. Create more concrete exception for your use case.

Reply:
Create API tests for this case please if you haven't already. It is an unhappy but necessary path

Reply:
Mira Ibai, nos hemos estado fijando con todo lo que nos has dicho y hay muchas cosas que o están repetidas (pero usadas en otros métodos) o incluso hay que tocar muchos tests y clases que no son de esta rama. Yo creo que si hacemos ese refactor ahora, lo demás quedaría raro o bien nos estariamos excediendo al propósito de esta rama. Lo tendremos en cuenta este mensaje para la siguiente rama que haremos que es un refactor general.

Saludos.

Reply:
ok